packages/DevTools/src/signature-editor/function-signature-editor.ts

```diff
@@ -3,285 +3,88 @@ import * as ui from 'datagrok-api/ui';
 import * as DG from 'datagrok-api/dg';
 import CodeMirror from 'codemirror';
 import { BehaviorSubject } from 'rxjs';
+import { DEFAULT_CATEGORY, FUNC_PROPS_FIELDS, tooltipMessage, obligatoryFuncProps, 
+  functionPropsLabels, highlightModeByLang, functionPropsCode, funcParamTypes, FUNC_PARAM_FIELDS, 
+  DIRECTION, functionParamsMapping, COMMON_TAG_NAMES, OPTIONAL_TAG_NAME, LANGUAGE, headerSign, 
+  optionTags, OPTIONAL_TAG_NAMES, getChoicesByName, FUNC_PROPS_FIELD, DATA_QUERY_VIEW } from './const';
 import 'codemirror/mode/javascript/javascript';
 import 'codemirror/mode/python/python';
 import 'codemirror/mode/octave/octave';
 import 'codemirror/mode/r/r';
 import 'codemirror/mode/julia/julia';
 import 'codemirror/mode/sql/sql.js';
-import '../css/styles.css';
+import '../../css/styles.css';
 
 export function functionSignatureEditor(view: DG.View) {
-  if (view.type == 'DataQueryView') 
-    addFseRibbonQuery(view);
-  else
-    addFseRibbonScript(view);
+  view.type === DATA_QUERY_VIEW ? addFseRibbonQuery(view) : addFseRibbonScript(view);
 }
 
 function addFseRibbonQuery(v: DG.View) {
   setTimeout(() => {
     //@ts-ignore
-    const iconFse = ui.iconFA('magic', () => openFse(v, v.root.lastChild.lastChild.parentElement.getElementsByClassName('CodeMirror cm-s-default')[0].CodeMirror.getDoc().getValue()), 'Open Signature Editor');
-    v.root.lastChild.lastChild.parentElement.querySelectorAll('.d4-tab-header-stripe')[0].appendChild(iconFse);
+    const iconFse = ui.iconFA('magic', () => openFse(v, v.root.querySelector('.CodeMirror').CodeMirror.getDoc().getValue()), 'Open Signature Editor');
+    //@ts-ignore
+    v.ribbonMenu.root.previousSibling.append(ui.div(iconFse,'d4-ribbon-item'));
   }, 500);
 }
 
 function addFseRibbonScript(v: DG.View) {
   setTimeout(() => {
     const panels = v.getRibbonPanels();
-    // @ts-ignore
-    const iconFse = ui.iconFA('magic', () => openFse(v, v.root.lastChild.lastChild.CodeMirror.getDoc().getValue()), 'Open Signature Editor');
+    //@ts-ignore
+    const iconFse = ui.iconFA('magic', () => openFse(v, v.root.querySelector('.CodeMirror').CodeMirror.getDoc().getValue()), 'Open Signature Editor');
```

Comment:
you can use `getElementsByClassName('CodeMirror)` instead of `querySelector`

Reply:
also no need for //ts-ignore, just cast it to `any` since this is exactly what's happening